shell/src/main/java/org/apache/kafka/shell/MetadataNodeManager.java

```diff
@@ -333,6 +357,36 @@ private void handleCommitImpl(MetadataRecordType type, ApiMessage message)
                 producerIds.create("nextBlockStartId").setContents(record.nextProducerId() + "");
                 break;
             }
+            case ACCESS_CONTROL_ENTRY_RECORD: {
+                AccessControlEntryRecord record = (AccessControlEntryRecord) message;
+                DirectoryNode acls = data.root.mkdirs("acl").mkdirs("id");
+                FileNode file = acls.create(record.id().toString());
+                file.setContents(AccessControlEntryRecordJsonConverter.write(record,
+                    AccessControlEntryRecord.HIGHEST_SUPPORTED_VERSION).toPrettyString());
+                break;
+            }
+            case REMOVE_ACCESS_CONTROL_ENTRY_RECORD: {
+                RemoveAccessControlEntryRecord record = (RemoveAccessControlEntryRecord) message;
+                DirectoryNode acls = data.root.mkdirs("acl").mkdirs("id");
+                acls.rmrf(record.id().toString());
+                break;
+            }
+            case FEATURE_LEVEL_RECORD: {
+                FeatureLevelRecord record = (FeatureLevelRecord) message;
+                DirectoryNode features = data.root.mkdirs("features");
+                if (record.featureLevel() == 0) {
+                    features.rmrf(record.name());
```

Comment:
Sorry, could you let me know where we mentioned `featureLevel == 0` means delete it? I checked the record schema, it didn't mention this:

https://github.com/apache/kafka/blob/4b310d1fe171d6d5edda13bb1baf2cf5a0d8eb68/metadata/src/main/resources/common/metadata/FeatureLevelRecord.json#L17-L26

Reply:
Firstly we have a `RemoveFeatureLevelRecord.json` but we removed it in #12207 and we use 0-0 to represent not supported: https://github.com/apache/kafka/blob/4b310d1fe171d6d5edda13bb1baf2cf5a0d8eb68/metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java#L290

Reply:
Thanks for sharing! What a magic 0, haha! Could we add a simple note into `FeatureLevelRecord.json`? Something like:
```
{ "name": "FeatureLevel", "type": "int16", "versions": "0+", 
     "about": "The current finalized feature level of this feature for the cluster. 0 means feature not enabled." } 
```

Reply:
Good suggestion.